# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.cpp
# Commit: 3367012c1d9c
# Full Hash: 3367012c1d9cbf581740171430f156cada8b68ee
# Author: stransky <stransky@redhat.com>
# Date: 2025-06-09 09:54:49
# Description:
#   Bug 1970275 [Wayland] Request Wayland event synchronication after explicit roundtrip r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D252868
# ==============================================================================

diff -r 7ee52ba8206f -r 3367012c1d9c widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp	Mon Jun 09 04:04:57 2025 +0000
+++ b/widget/gtk/nsWaylandDisplay.cpp	Mon Jun 09 05:37:33 2025 +0000
@@ -685,8 +685,8 @@
   mFormats = new DMABufFormats();
   mRegistry = wl_display_get_registry(mDisplay);
   wl_registry_add_listener(mRegistry, &registry_listener, this);
+  wl_display_roundtrip(mDisplay);
   RequestAsyncRoundtrip();
-  wl_display_roundtrip(mDisplay);
   WaitForAsyncRoundtrips();
   EnsureDMABufFormats();
 
